src/Groups/matrices/MatGrp.jl

```diff
@@ -561,9 +561,11 @@ function compute_order(G::MatrixGroup{T}) where {T <: Union{nf_elem, fmpq}}
     # The call to `IsHandledByNiceMonomorphism` triggers an expensive
     # computation of `IsFinite` which we avoid by checking
     # `HasIsHandledByNiceMonomorphism` first.
-    return fmpz(GAPWrap.Order(G.X))
+    return fmpz(GAPWrap.Size(G.X))
   else
-    order(isomorphic_group_over_finite_field(G)[1])
+    n = order(isomorphic_group_over_finite_field(G)[1])
+    GAP.Globals.SetSize(G.X, GAP.Obj(n))
```

Comment:
I would suggest `set_order(G, n)`.
However, currently `set_order` is wrong; it should call `GAP.Globals.SetSize` with second argument `GAP.Obj(val)` not with `GapObj(val)`.

Reply:
Yeah, and I found a few more things. I will now also set up a NiceMonomorphism so that GAP can do even more things

Reply:
also, if the group is infinite then `isomorphic_group_over_finite_field` also determines this, I'll make sure we also use that information

Reply:
In this PR here or in the next one?